presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -480,9 +480,6 @@ public TableMetadata getTableMetadata(Session session, TableHandle tableHandle)
         CatalogName catalogName = tableHandle.getCatalogName();
         ConnectorMetadata metadata = getMetadata(session, catalogName);
         ConnectorTableMetadata tableMetadata = metadata.getTableMetadata(session.toConnectorSession(catalogName), tableHandle.getConnectorHandle());
-        if (tableMetadata.getColumns().isEmpty()) {
-            throw new PrestoException(NOT_SUPPORTED, "Table has no columns: " + tableHandle);
```

Comment:
How will `SELECT *` work for such a table?
Do we have a test for this?

(IIRC i added the check here to prevent obscure errors later on, but this may be outdated.)